lib/model/channel.dart

```diff
@@ -199,6 +199,51 @@ class ChannelStoreImpl with ChannelStore {
           streamsByName.remove(stream.name);
           subscriptions.remove(stream.streamId);
         }
+
+      case ChannelUpdateEvent():
+        final ChannelUpdateEvent(:streamId, name: String streamName) = event;
+        assert(identical(streams[streamId], streamsByName[streamName]));
+        assert(subscriptions[streamId] == null
+          || identical(subscriptions[streamId], streams[streamId]));
```

Comment:
This looks to me like the kind of `assert` we do use: checking that our own data structures are consistent with each other.